cephadm-clients.yml

```
@@ -0,0 +1,172 @@
+
+    - name: fail if client_group is NOT in the inventory
+      fail:
+        msg: >
+          Variable client_group '{{ client_group }}' is not defined in the inventory
+      when: client_group not in groups
+
+    - name: fail if keyring variable is missing
+      fail:
+        msg: |
+          You must supply a 'keyring' variable that defines the path to the key
+          that you want to distribute to your client machines
+      when: keyring is not defined
+
+    - name: create the directory for the client payload (if needed)
+      file:
+        path: "{{ client_conf_dir }}"
+        state: directory
+
+
+- name: Confirm admin host is ready
+  hosts: admin[0]
+  become: yes
+  tasks:
+
+    - name: check fsid is present on {{ inventory_hostname }}
+      stat:
+        path: /var/lib/ceph/{{ fsid }}
+      register: fsid_stat
+
+    - name: fail if fsid is not present
+      fail:
+        msg: >
+          The given fsid ({{ fsid }}), is not present in /var/lib/ceph on {{ inventory_hostname }}
+      when:
+        - not fsid_stat.stat.exists | bool
+        - not fsid_stat.stat.isdir | bool
+
+    - name: check keyring status on {{ inventory_hostname }}
+      stat:
+        path: "{{ keyring }}"
+      register: keyring_stat
+
+    - name: fail if keyring not found on {{ inventory_hostname }}
+      fail:
+        msg: >
+          The keyring path provided '{{ keyring }}' can not be found on {{ inventory_hostname }}
+      when: not keyring_stat.stat.exists | bool
+
+    - name: check conf is OK to use
+      stat:
+        path: "{{ conf }}"
+      register: conf_stat
+      when: conf is defined
+
+    - name: fail if conf supplied is not on {{ inventory_hostname }}
+      fail:
+        msg: |
+          The conf file '{{ conf }}' can not be found on {{ inventory_hostname }}
+      when:
+        - conf is defined
+        - not conf_stat.stat.exists | bool
+        - not conf_stat.stat.isreg | bool
+
+
+- name: Assemble client payload on the local machine
+  hosts: admin[0]
+  become: yes
+  tasks:
+
+    - import_role:
+        name: ceph-defaults
+
+    - name: Fetch the keyring to /tmp on the ansible machine
+      fetch:
+        src: "{{ keyring }}"
+        dest: "{{ client_conf_dir }}/ceph.keyring"
+        flat: yes
+
+    - name: Fetch the conf if it's supplied
+      fetch:
+        src: "{{ conf }}"
+        dest: "{{ client_conf_dir }}/ceph.conf"
+        flat: yes
+      when:
+        - conf is defined
+        - conf | length > 0
+
+    - name: create minimal conf as a default
+      shell:
+        cmd: "cephadm shell -- ceph config generate-minimal-conf > /etc/ceph/minimal.conf"
+      when: conf is undefined
+
+    - name: Fetch the minimal ceph.conf
+      fetch:
+        src: /etc/ceph/minimal.conf
+        dest: "{{ client_conf_dir }}/ceph.conf"
+        flat: yes
+      when: conf is undefined
+
+
+- name: Distribute client configuration
+  hosts: "{{ client_group }}"
```

I'm not sure we need to gather facts here?

Gathering facts in a large cluster deployment with a lot of client nodes may be very time and/or memory-consuming.
 
```suggestion
  hosts: "{{ client_group }}"
  gather_facts: false
```

---

ACK. Will fix